### DISCUSSION TITLE
Add libslirp on melonds for the WiFi

### DISCUSSION THREAD

**Discussion by Swappo91** [rafaelvcaetano/melonDS-android#1053] [Ideas]
Hi bro! First of all thank you! I love to test all nighty builds. You are doing all great. Emulstor is smooth and stable. I have only this question. Now melonds official have 0.9.5 i think its time to get the possibility to all android player to go online (people are waiting to play mk ds or pokemons online.

Is too hard to add libslirp on your melonds?

**Comment by rafaelvcaetano** [comment]
libslirp is now available on the latest nightly builds. Give it a try!

**Reply by Swappo91** [reply]
of course!

**Reply by Swappo91** [reply]
Okay tested. Seems work good, but i have an issue. 

1 time i try to connect, and melonds have crash.
Now when i try to restart and open the game, they continue to crash at launch of the rom.

**Reply by rafaelvcaetano** [reply]
Does it really crash? Or does it just freeze?

**Reply by Swappo91** [reply]
Crash crash. The second time you try to open melonds they crash every time you try to load the rom

**Reply by rafaelvcaetano** [reply]
Can you tell me which ROM you are using?

**Reply by Swappo91** [reply]
pokemon gold italian version. That is the bug

**Reply by Swappo91** [reply]
how i can try to send you the log?

**Comment by Swappo91** [comment]
![Uploading Screenshot_2023-08-12-15-52-24-090_com.miui.bugreport.jpg…]()

**Comment by Swappo91** [comment]
![Screenshot_2023-08-12-15-52-21-056_com miui bugreport](https://github.com/rafaelvcaetano/melonDS-android/assets/105309275/b1ff2ccf-a22b-443a-bb14-874798aa6931)

**Reply by Swappo91** [reply]
maybe another thing can help you. I need to unistall/reinstall melonds app to reload the rom. I think the problem is not on wfc libsript update. But a file problem. Save files maybe? config files.

If you need addition help just write here

**Reply by rafaelvcaetano** [reply]
From the crash log you sent here, it's definitely not related to Wi-Fi. It seems to be something wrong in a custom layout.

**Reply by Swappo91** [reply]
its possible a solution?

**Reply by Swappo91** [reply]
another log when i try to edit the imputer layout

**Comment by Swappo91** [comment]
![Uploading Screenshot_2023-08-12-16-28-43-809_com.miui.bugreport.jpg…]()

**Comment by Swappo91** [comment]
![Screenshot_2023-08-12-16-28-46-625_com miui bugreport](https://github.com/rafaelvcaetano/melonDS-android/assets/105309275/f5f31747-ea7e-4bc1-9e56-4250304a1afc)

**Comment by Swappo91** [comment]
another info, my device is xiaomi 12.
Issue is only for custom layouts. i need to custom the layout because in my screen resolution is impossjble play well with default layouts, the are so small and located on the end of the screen

**Reply by rafaelvcaetano** [reply]
The issue should be fixed in the latest nightly release.

**Reply by Swappo91** [reply]
tested, still the issue i add the logs here

**Comment by Swappo91** [comment]
![Uploading Screenshot_2023-08-13-14-26-13-038_com.miui.bugreport.jpg…]()

**Comment by Swappo91** [comment]
![Uploading Screenshot_2023-08-13-14-26-11-110_com.miui.bugreport.jpg…]()

**Reply by Swappo91** [reply]
update: seems fixed, im continue testing. thanks
